### PR TITLE
Fix "Trying to access array offset on null" warning

### PR DESCRIPTION
**Pull request by xcopy**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌

**Note:** this does not solve the whole problem described in the issue #17365, but only part of it.

### CODE REVIEW

**Comment by xcopy** [comment]
Btw, can find more similar block codes, i.e. 

```php
$error = error_get_last();
Yii::warning("...: {$error['message']}", __METHOD__);
```

and I guess the same kind of problem might arise with them.

However, I like the approach in the class `yii\web\Session`:

https://github.com/yiisoft/yii2/blob/49cfd3b767d33cfdb3808c4bceb7caba963133ad/framework/web/Session.php#L163-L165

**Review by samdark — APPROVED** [review]
:+1: Would you please add a line for CHANGELOG? Thanks.

**Comment by xcopy** [comment]
> Btw, can find more similar code-block, i.e.
> 
> ```
> $error = error_get_last();
> Yii::warning("...: {$error['message']}", __METHOD__);
> ```
> 
> and I guess the same kind of problem might arise with them.
> 
> However, I like the approach in the class `yii\web\Session`:
> 
> https://github.com/yiisoft/yii2/blob/49cfd3b767d33cfdb3808c4bceb7caba963133ad/framework/web/Session.php#L163-L165

@samdark how about this one? Should fix 'em all?

**Comment by xcopy** [comment]
> 👍 Would you please add a line for CHANGELOG? Thanks.

Done ✅

**Comment by samdark** [comment]
Worth fixing all these if you're sure the issue could occur.

**Review by Arhell — APPROVED** [review]

**Review by bizley — APPROVED** [review]

**Comment by bizley** [comment]
LGTM although I prefer classic ifs.

File: framework/caching/FileCache.php [hunk 1]
```diff
@@ -158,8 +158,11 @@ protected function setValue($key, $value, $duration)
             return @touch($cacheFile, $duration + time());
         }
 
-        $error = error_get_last();
-        Yii::warning("Unable to write cache file '{$cacheFile}': {$error['message']}", __METHOD__);
+        $message = "Unable to write cache file '{$cacheFile}'";
```
Comment (mtangoo) [inline framework/caching/FileCache.php:161]: Use Yii::t so that it can be translated to multiple languages

File: framework/caching/FileCache.php [hunk 2]
```diff
@@ -265,20 +268,22 @@ protected function gcRecursive($path, $expiredOnly)
...
                 }
                 $fullPath = $path . DIRECTORY_SEPARATOR . $file;
+                $message = null;
                 if (is_dir($fullPath)) {
                     $this->gcRecursive($fullPath, $expiredOnly);
                     if (!$expiredOnly) {
                         if (!@rmdir($fullPath)) {
-                            $error = error_get_last();
-                            Yii::warning("Unable to remove directory '{$fullPath}': {$error['message']}", __METHOD__);
+                            $message = "Unable to remove directory '$fullPath'";
```
Comment (mtangoo) [inline framework/caching/FileCache.php:276]: Yii::t again for non-english users

File: framework/log/FileTarget.php [hunk 3]
```diff
@@ -131,8 +131,9 @@ public function export()
         }
         $writeResult = @fwrite($fp, $text);
         if ($writeResult === false) {
-            $error = error_get_last();
-            throw new LogRuntimeException("Unable to export log through file ({$this->logFile})!: {$error['message']}");
+            $message = "Unable to export log through file ($this->logFile)!";
```
Comment (mtangoo) [inline framework/log/FileTarget.php:134]: @samdark is there any reason why Yii::t was not used in error messages? 
Should it be left that way or addressed?

**Comment by xcopy** [comment]
> LGTM although I prefer classic ifs.

Done ✅

File: framework/log/FileTarget.php (hunk 3 above)
Comment (rob006) [reply framework/log/FileTarget.php:134]: `Yii::t()` is used only for messages and exceptions that should be displayed to end user (for example HTTP exceptions). I don't thing that these messages should be translated.

Comment (mtangoo) [reply framework/log/FileTarget.php:134]: @xcopy never mind. @rob006 yeah, I was wondering but your comment makes sense.

**Comment by mtangoo** [comment]
I think we are good to go. @xcopy thanks for your time and effort. If you feel you can tackle another PR, feel free to do so!

**Pull request merged**